### ISSUE TITLE
Sanpark skin main page has login issue

### ISSUE THREAD

**Issue by mbjones** [NCEAS/metacat#795]
---

Author Name: **Jing Tao** (Jing Tao)
Original Redmine Issue: 5533, https://projects.ecoinformatics.org/ecoinfo/issues/5533
Original Date: 2011-11-04
Original Assignee: Jing Tao

---

Hi Jing

Yes I see that it works on firefox the problem is that all our staff have to use IE So is there a way to fix it

Regards

Judith

 

From: jing [mailto:tao@nceas.ucsb.edu]
Sent: 03 November 2011 11:44 PM
To: Judith Botha
Subject: Re: WEb page error

 

Hi, Judith:

Can you try firefox? I tried the firefox and I can see "Welcome, Jing Tao...." even i logged in dataknp's home page.

Thanks,

Jing

On 11/02/2011 11:52 PM, Judith Botha wrote:

 

Hi Jung

When I open dataknp and use the login on the page with IE it says logging in and then nothing happens after that. It seems like it does not log you in at all as if I then do a search I only get the datapackages that are public. If I go to register data and login through that interface then when  I go back to the repository home it shows the welcome Judith. Kruger and then when I search I get the datapackages that I have permission to access. Please can you have a look at why that is happening

Thanks

J

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Jing Tao** (Jing Tao)
Original Date: 2011-11-08T16:19:15Z

---

An additional message from judith:

Jing

I notice that when you login through the web page on firefox You get the message that you are logged in but actually you aren’t as you can then still not see the datapackages that are not public

Thanks Jing

Regards

J

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Jing Tao** (Jing Tao)
Original Date: 2011-11-17T20:03:03Z

---

For the first issue that a user could not log in on IE, the reason is IE invalid the login form after a div, which contains the form, was changed the content (Put label "logging in"). So the solution is: serializing the form to a variable before changing the content of the div. Send the serialized form data to the metacat server.

The second issue is that a user can log in on FF, but the search only showed the public readable data. We figured out that the search form using request.getSession().getId() and there was no mechanism to update the session id after login; so the search form would still use public session id. We add a reloading mechanism after login, then the search works (even on IE).

The involved files are

lib/style/skins/sanparks/sanparksLogin.js
lib/style/skins/sanparks/sanparksLogin.jsp
lib/style/common/ajax-utils.js
